<?xml version="1.0"?>
<launch>
  <rosparam command="load" file="$(find roscopter)/params/ekf_gains.yaml" ns="ekf_node"/>
  <rosparam command="load" file="$(find roscopter)/params/ekf_drone.yaml" ns="ekf_node"/>

  <node name="ekf_node" type="ekf_node" pkg="roscopter" output="screen">
    <remap from="imu" to="/imu/data"/>
    <remap from="truth/pose" to="/vrpn_client_node/planck/pose"/>
  </node>

  <!--Note: You should run the motion capture node on the motion capture computer-->
  <!--motion capture-->
  <!--<node name="vrpn" type="vrpn_client_node" pkg="vrpn_client_ros" output="screen">-->
    <!--<param name="refresh_tracker_frequency" value="1"/>-->
    <!--<param name="use_server_time" value="true"/>-->
    <!--<param name="server" value="192.168.1.186"/>-->
    <!--<remap from="vrpn/planck/pose" to="truth/pose"/>-->
    <!--</node>-->

  <!-- FCU communication -->
  <node name="rosflight_io" pkg="rosflight" type="rosflight_io" output="screen"/>

  <!-- Load common parameters -->
  <rosparam command="load" file="$(find roscopter)/params/planck_450.yaml"/>

  <!-- PID Position Controller -->
  <node name="controller" pkg="roscopter" type="controller">
    <remap from="estimate" to="odom"/>
    <!-- <remap from="estimate" to="estimate"/> -->
  </node>

  <!--Car Follower High Level command-->
  <node name="car_follower" pkg="roscopter" type="car_follower.py" output="screen">
    <remap from="car_mocap" to="/vrpn_client_node/car1/pose"/>
    <remap from="high_level_command" to="high_level_command"/>
  </node>

</launch>
